varnishkafka.c

```diff
@@ -1349,6 +1349,48 @@ static int kafka_stats_cb (rd_kafka_t *rk_arg UNUSED, char *json, size_t json_le
 	return 0;
 }
 
+static int kafka_dumpstats_cb (rd_kafka_t *rk_arg UNUSED, char *json, size_t json_len UNUSED,
+			    void *opaque UNUSED) {
+
+	int status;
+
+	if (!(conf.stats_fp = fopen(conf.stats_file, "w"))) {
+		vk_log("STATS", LOG_ERR,
+			"Failed to open statistics log file %s: %s\n",
+			conf.stats_file, strerror(errno));
+		return 0;
+	}
+
+	status = fprintf(conf.stats_fp, "{ \"kafka\": %s, \"varnishkafka\": { "
+		"\"time\":%llu, "
+		"\"tx\":%"PRIu64", "
+		"\"txerr\":%"PRIu64", "
+		"\"kafka_drerr\":%"PRIu64", "
+		"\"trunc\":%"PRIu64", "
+		"\"seq\":%"PRIu64" "
+		"} }\n",
+		json,
+		(unsigned long long)time(NULL),
+		cnt.tx,
+		cnt.txerr,
+		cnt.kafka_drerr,
+		cnt.trunc,
+		conf.sequence_number);
+
+	if (status < 0) {
+		vk_log("STATS", LOG_ERR,
+			"Failed to write to statistics log file %s: %s\n",
+			conf.stats_file, strerror(errno));
+		fclose(conf.stats_fp);
```

Comment:
I realized only now, but, you write both varnishkafka statistics (print_stats callback) and librdkafka statistics (kafka_stats_cb callback) to the same file and truncate it. Most likely it will become impossible to read one of these two stats.